llvm/test/Transforms/AggressiveInstCombine/AArch64/or-load.ll

```diff
@@ -1698,6 +1718,302 @@ define i32 @loadCombine_4consecutive_mixsize2(ptr %p) {
+; ALL-NEXT:    [[L3:%.*]] = load i4, ptr [[P2]], align 1
+; ALL-NEXT:    [[E1:%.*]] = zext i8 [[L1]] to i16
+; ALL-NEXT:    [[E2:%.*]] = zext i4 [[L2]] to i16
+; ALL-NEXT:    [[E3:%.*]] = zext i4 [[L3]] to i16
+; ALL-NEXT:    [[S2:%.*]] = shl i16 [[E2]], 8
+; ALL-NEXT:    [[S3:%.*]] = shl i16 [[E3]], 12
+; ALL-NEXT:    [[O1:%.*]] = or i16 [[E1]], [[S2]]
+; ALL-NEXT:    [[O2:%.*]] = or i16 [[O1]], [[S3]]
+; ALL-NEXT:    ret i16 [[O2]]
+;
+  %p1 = getelementptr i4, ptr %p, i32 2
+  %p2 = getelementptr i4, ptr %p, i32 3
+  %l1 = load i8, ptr %p
+  %l2 = load i4, ptr %p1
+  %l3 = load i4, ptr %p2
+
+  %e1 = zext i8 %l1 to i16
+  %e2 = zext i4 %l2 to i16
+  %e3 = zext i4 %l3 to i16
+
+  %s2 = shl i16 %e2, 8
+  %s3 = shl i16 %e3, 12
+
+  %o1 = or i16 %e1, %s2
+  %o2 = or i16 %o1, %s3
+  ret i16 %o2
+}
+
+define i32 @loadCombine_2consecutive_mixsize_not_equal_store_size(ptr %p) {
+; ALL-LABEL: @loadCombine_2consecutive_mixsize_not_equal_store_size(
+; ALL-NEXT:    [[P1:%.*]] = getelementptr i4, ptr [[P:%.*]], i32 1
+; ALL-NEXT:    [[L1:%.*]] = load i4, ptr [[P]], align 1
+; ALL-NEXT:    [[L2:%.*]] = load i28, ptr [[P1]], align 4
+; ALL-NEXT:    [[E1:%.*]] = zext i4 [[L1]] to i32
+; ALL-NEXT:    [[E2:%.*]] = zext i28 [[L2]] to i32
+; ALL-NEXT:    [[S2:%.*]] = shl i32 [[E2]], 4
+; ALL-NEXT:    [[O1:%.*]] = or i32 [[E1]], [[S2]]
+; ALL-NEXT:    ret i32 [[O1]]
+;
+  %p1 = getelementptr i4, ptr %p, i32 1
+  %l1 = load i4, ptr %p
+  %l2 = load i28, ptr %p1
+  %e1 = zext i4 %l1 to i32
+  %e2 = zext i28 %l2 to i32
+  %s2 = shl i32 %e2, 4
+  %o1 = or i32 %e1, %s2
+  ret i32 %o1
+}
+
+define i32 @loadCombine_2consecutive_mixsize_not_equal_store_size2(ptr %p) {
+; ALL-LABEL: @loadCombine_2consecutive_mixsize_not_equal_store_size2(
+; ALL-NEXT:    [[P1:%.*]] = getelementptr i4, ptr [[P:%.*]], i32 7
+; ALL-NEXT:    [[L1:%.*]] = load i28, ptr [[P]], align 4
+; ALL-NEXT:    [[L2:%.*]] = load i4, ptr [[P1]], align 1
+; ALL-NEXT:    [[E1:%.*]] = zext i28 [[L1]] to i32
+; ALL-NEXT:    [[E2:%.*]] = zext i4 [[L2]] to i32
+; ALL-NEXT:    [[S2:%.*]] = shl i32 [[E2]], 28
+; ALL-NEXT:    [[O1:%.*]] = or i32 [[E1]], [[S2]]
+; ALL-NEXT:    ret i32 [[O1]]
+;
+  %p1 = getelementptr i4, ptr %p, i32 7
+  %l1 = load i28, ptr %p
+  %l2 = load i4, ptr %p1
+  %e1 = zext i28 %l1 to i32
+  %e2 = zext i4 %l2 to i32
+  %s2 = shl i32 %e2, 28
+  %o1 = or i32 %e1, %s2
+  ret i32 %o1
+}
+
+define i32 @loadCombine_2consecutive_mixsize_not_equal_store_size3(ptr %p) {
+; ALL-LABEL: @loadCombine_2consecutive_mixsize_not_equal_store_size3(
+; ALL-NEXT:    [[P1:%.*]] = getelementptr i1, ptr [[P:%.*]], i32 23
+; ALL-NEXT:    [[L1:%.*]] = load i23, ptr [[P]], align 4
+; ALL-NEXT:    [[L2:%.*]] = load i9, ptr [[P1]], align 2
+; ALL-NEXT:    [[E1:%.*]] = zext i23 [[L1]] to i32
+; ALL-NEXT:    [[E2:%.*]] = zext i9 [[L2]] to i32
+; ALL-NEXT:    [[S2:%.*]] = shl i32 [[E2]], 24
+; ALL-NEXT:    [[O1:%.*]] = or i32 [[E1]], [[S2]]
+; ALL-NEXT:    ret i32 [[O1]]
+;
+  %p1 = getelementptr i1, ptr %p, i32 23
```

Comment:
```suggestion
  %p1 = getelementptr i23, ptr %p, i32 1
```

Reply:
What you wrote is a 23-byte (not bit) GEP, which is probably not what you intended.